grails-app/domain/com/unifina/domain/security/IntegrationKey.groovy

```diff
@@ -26,6 +26,7 @@ class IntegrationKey implements Serializable {
 	}
 
 	static constraints = {
+		idInService unique: true
```

Comment:
- Requires new migrations to be generated with the `db_diff` util, so that this will update the database. For that migration to succeed, the database must be cleaned of such occurrences.
- If we add this constraint, then the migration changes are actually unnecessary because duplicate `id_in_service` will be prevented on database level. Of course, the migration can remain as it is now, as the current migration will work in both cases (allow duplicates vs. demand unique).

Adding this constraint is probably the way to go. To apply this change, manual cleaning of the production database is required, and then a new dump from prod to staging must happen to make this work in staging.